openmc/deplete/nuclide.py

```diff
@@ -0,0 +1,570 @@
+                sum_yield = fission_yield.yields.sum()
+                stat = 2.0 - tolerance <= sum_yield <= 2.0 + tolerance
+                if stat:
+                    continue
+                msg = msg_func(
+                    name=self.name, actual=sum_yield,
+                    expected=2.0, tol=tolerance,
+                    prop="fission yields (E = {:7.4e} eV)".format(energy))
+                if strict:
+                    raise ValueError(msg)
+                elif quiet:
+                    return False
+                warn(msg)
+                valid = False
+
+        return valid
+
+
+class FissionYieldDistribution(Mapping):
+    """Energy-dependent fission product yields for a single nuclide
+
+    Can be used as a dictionary mapping energies and products to fission
+    yields::
+
+        >>> fydist = FissionYieldDistribution{
+        ...     {0.0253: {"Xe135": 0.021}})
+        >>> fydist[0.0253]["Xe135"]
+        0.021
+
+    Parameters
+    ----------
+    fission_yields : dict
+        Dictionary of energies and fission product yields for that energy.
+        Expected to be of the form ``{float: {str: float}}``. The first
+        float is the energy, typically in eV, that represents this
+        distribution. The underlying dictionary maps fission products
+        to their respective yields.
+
+    Attributes
+    ----------
+    energies : tuple
+        Energies for which fission yields exist. Sorted by
+        increasing energy
+    products : tuple
+        Fission products produced at all energies. Sorted by name.
+    yield_matrix : numpy.ndarray
+        Array ``(n_energy, n_products)`` where
+        ``yield_matrix[g, j]`` is the fission yield of product
+        ``j`` for energy group ``g``.
+
+    See Also
+    --------
+    * :meth:`from_xml_element` - Construction methods
+    * :class:`FissionYield` - Class used for storing yields at a given energy
+    """
+
+    def __init__(self, fission_yields):
+        # mapping {energy: {product: value}}
+        energies = sorted(fission_yields)
+
+        # Get a consistent set of products to produce a matrix of yields
+        shared_prod = set.union(*(set(x) for x in fission_yields.values()))
+        ordered_prod = sorted(shared_prod)
+
+        yield_matrix = empty((len(energies), len(shared_prod)))
+
+        for g_index, energy in enumerate(energies):
+            prod_map = fission_yields[energy]
+            for prod_ix, product in enumerate(ordered_prod):
+                yield_val = prod_map.get(product)
+                yield_matrix[g_index, prod_ix] = (
+                    0.0 if yield_val is None else yield_val)
+        self.energies = tuple(energies)
+        self.products = tuple(ordered_prod)
+        self.yield_matrix = yield_matrix
+
+    def __len__(self):
+        return len(self.energies)
+
+    def __getitem__(self, energy):
+        if energy not in self.energies:
```

Comment:
For completeness, you may want to check that the appropriate element exists in `yield_matrix` as well as `energies`.